### PR TITLE
Add yarn support

### PR DESCRIPTION
**Pull request by gish**

### CODE REVIEW

**Comment by gish** [comment]
Travis is configured to run node v0.12. Seems like that version is not supported by yarn. Can Travis be configured to run a newer version?

Edit: Figured it out!

File: package.json [hunk 1]
```diff
@@ -8,15 +8,15 @@
...
     "test:debug": "karma start ./test/karma.conf.js --no-single-run --reporters mocha --browsers Chrome --source-map",
     "test:tdd": "karma start ./test/karma.conf.js --auto-watch --no-single-run ",
-    "validate": "npm run lint && npm test",
-    "prebuild": "npm run validate",
-    "prepush": "npm run validate",
-    "prepublish": "npm run build",
+    "validate": "run-s lint test",
+    "prebuild": "run-s validate",
+    "prepush": "run-s validate",
+    "prepublish": "run-s build",
```
Comment (iamstarkov) [inline package.json:19]: do even need `npm-run-all` after this refactoring? i mean primary use case is go run several npm scripts either sequentially or in parallel. in our case we run single scripts

Comment (gish) [reply package.json:19]: We might not. I assumed it was needed to be yarn/npm agnostic.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,9 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## Upcoming
+* Add yarn support
```
Comment (iamstarkov) [inline CHANGELOG.md]: is yarn internal business of the library or in other words, does it affect api surface of the library?

Comment (iamstarkov) [reply CHANGELOG.md]: i would also add `isomorphic-fetch` unlocking

File: package.json (hunk 1 above)
Comment (iamstarkov) [reply package.json:19]: > If a script is invoked with Yarn, npm-run-all will correctly use Yarn to execute the plan's child scripts.

File: CHANGELOG.md (hunk 2 above)
Comment (gish) [reply CHANGELOG.md]: Not the library per se, only dev experience.

File: package.json (hunk 1 above)
Comment (gish) [reply package.json:19]: And what if it's invoked with npm?

Comment (iamstarkov) [reply package.json:19]: scripts in turn will be invoked with npm

**Comment by bstream** [comment]
👏

**Pull request merged**